### PR TITLE
Feature kafka monitoring

### PR DESCRIPTION
**Pull request by ZhuWang1112**
- [ ] If this is non-trivial feature, paste the links/URLs to the design doc.

- [x] Update the documentation to include this new feature.

- [x] Tests(including UT, IT, E2E) are added to verify the new feature.

- [x] If it's UI related, attach the screenshots below.

- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.

- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

Kafka Root Dashboard
kafka-root

Kafka Cluster Dashboard
kafka-cluster

clusterOverView

Kafka Broker Dashboard
kafka-broker0
kafka-broker1

### CODE REVIEW

File: test/e2e-v2/cases/kafkamq/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+version: '3'
+
+services:
```
Comment (wankai123) [inline test/e2e-v2/cases/kafkamq/docker-compose.yml:18]: missing oap service deployment

File: test/e2e-v2/cases/kafkamq/otel-collector-config.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+              target_label: broker
+              replacement: $$1
+
+processors:
+  batch:
+
+exporters:
+  otlp:
+    # endpoint: "10.0.0.120:11800" # The OAP Server address
+    endpoint: "192.168.0.120:11800"
```
Comment (wankai123) [inline test/e2e-v2/cases/kafkamq/otel-collector-config.yaml:41]:
```suggestion
    # The OAP Server address
    endpoint: oap:11800
```

**Comment by wankai123** [comment]
Please don't forget the document.

**Comment by wu-sheng** [comment]
1. skywalking-ui submodule seems not to be updated correctly. You should not change the commit ID of this submodule. Please align with the master branch.
2. Your newly added e2e is not in the CI pipeline, so it would not run even if I approved this. Please add them to the proper CI control file.

If you are not sure what I meant, please connect with your mentor. Thanks.

**Comment by wu-sheng** [comment]
`Conflicting files` must be resolved, otherwise, CI can't run.

**Comment by ZhuWang1112** [comment]
> `Conflicting files` must be resolved, otherwise, CI can't run.

**Comment by wu-sheng** [comment]
Why close?

**Comment by ZhuWang1112** [comment]
> `Conflicting files` must be resolved, otherwise, CI can't run.
Thanks!  I'm still revising some previous content. There are still some problems with e2e's current local testing. I will fix the problem you mentioned in time

**Comment by ZhuWang1112** [comment]
> Why close?
Sorry for the mistake.

File: oap-server/server-starter/src/main/resources/application.yml [hunk 3]
```diff
@@ -393,7 +393,7 @@ query:
     # abort a query if the total number of data fields queried exceeds the defined threshold.
     maxQueryComplexity: ${SW_QUERY_MAX_QUERY_COMPLEXITY:3000}
     # Allow user add, disable and update UI template
-    enableUpdateUITemplate: ${SW_ENABLE_UPDATE_UI_TEMPLATE:false}
+    enableUpdateUITemplate: ${SW_ENABLE_UPDATE_UI_TEMPLATE:true}
```
Comment (wuwen5) [inline oap-server/server-starter/src/main/resources/application.yml:396]: Why modify the default values? Please confirm if this modification is related to this feature.
If necessary, please modify the corresponding document. (`skywalking/docs/en/setup/backend/Configuration vocabulary.md`)

Comment (wu-sheng) [reply oap-server/server-starter/src/main/resources/application.yml:396]: This should not open by default.
I believe this is a local change, please revert.

**Comment by wu-sheng** [comment]
If you are ready, please make this as `ready for review`. I would approve your tests on CI.

**Comment by wu-sheng** [comment]
> test/e2e-v2/cases/kafka/kafka-monitoring/jmx_prometheus_javaagent-0.18.0.jar

This can't be accepted. There is no binary jar allowed in ASF source code repository and final release.
If you need to it, please download in the test stage.

**Comment by ZhuWang1112** [comment]
> If you are ready, please make this as `ready for review`. I would approve your tests on CI.

Thank you so much! I fixed the download method of javagent.jar and have opened this PR

File: oap-server/server-starter/src/main/resources/application.yml (hunk 3 above)
Comment (ZhuWang1112) [reply oap-server/server-starter/src/main/resources/application.yml:396]: Sorry, this is just for local testing convenience. I will turn it off before the final version

Comment (wu-sheng) [reply oap-server/server-starter/src/main/resources/application.yml:396]: The best practice is, you should revert this, and change the local system environment variable to set as true.

**Comment by wu-sheng** [comment]
Is the last commit breaking something? Most tests passed, but fail this time.

**Comment by ZhuWang1112** [comment]
> Is the last commit breaking something? Most tests passed, but fail this time.

 I thought I didn't make changes to the other codes. I only added the missing license headers, removed a YML file to download it during build time, closed the enableUpdateUITemplate, and cleaned up some comments. I noticed that the Kafka e2e tests I wrote passed successfully.

**Comment by wu-sheng** [comment]
Let me rerun cases.

File: oap-server/server-starter/src/main/resources/otel-rules/kafka/kafka-cluster.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    exp: kafka_server_replicafetchermanager_maxlag.sum(['cluster', 'broker'])
+
+
+
+
+
+
+
+
+
```
Comment (wankai123) [inline oap-server/server-starter/src/main/resources/otel-rules/kafka/kafka-cluster.yaml:67]:
```suggestion
    exp: kafka_server_replicafetchermanager_maxlag.sum(['cluster', 'broker'])

```

File: test/e2e-v2/cases/kafka/kafka-monitoring/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+      - ALLOW_PLAINTEXT_LISTENER=yes
+      - KAFKA_OPTS=-javaagent:/etc/jmx_prometheus_javaagent-0.18.0.jar=7071:/etc/kafka-2_0_0.yml
+    depends_on:
+      zookeeper:
+        condition: service_healthy
+    # healthcheck:
+    #   test: ["CMD", "kafka-topics.sh", "--list", "--zookeeper", "zookeeper:2181"]
+    #   interval: 5s
+    #   timeout: 60s
+    #   retries: 120
```
Comment (wankai123) [inline test/e2e-v2/cases/kafka/kafka-monitoring/docker-compose.yml:67]:
```suggestion
    health check:
      test: ["CMD", "kafka-topics.sh", "--list", "--zookeeper", "zookeeper:2181"]
      interval: 5s
      timeout: 60s
      retries: 120
```

File: test/e2e-v2/cases/kafka/kafka-monitoring/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+    environment:
+      - KAFKA_ENABLE_KRAFT=no
+      - KAFKA_CFG_ZOOKEEPER_CONNECT=zookeeper:2181
+      - KAFKA_CFG_LISTENERS=PLAINTEXT://:9093
+      - KAFKA_CFG_ADVERTISED_LISTENERS=PLAINTEXT://broker2:9093
+      - ALLOW_PLAINTEXT_LISTENER=yes
+      - KAFKA_OPTS=-javaagent:/etc/jmx_prometheus_javaagent-0.18.0.jar=7072:/etc/kafka-2_0_0.yml
+    depends_on:
+      zookeeper:
+        condition: service_healthy
```
Comment (wankai123) [inline test/e2e-v2/cases/kafka/kafka-monitoring/docker-compose.yml:89]:
```suggestion
        condition: service_healthy
    health check:
      test: ["CMD", "kafka-topics.sh", "--list", "--zookeeper", "zookeeper:2181"]
      interval: 5s
      timeout: 60s
      retries: 120
```

File: oap-server/server-starter/src/main/resources/ui-initialized-templates/kafka/kafka-cluster.json [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+				"name": "Overview",
+				"children": [{
+					"x": 16,
+					"y": 10,
+					"w": 8,
+					"h": 10,
+					"i": "13",
+					"type": "Widget",
+					"metricTypes": ["readLabeledMetricsValues"],
+					"metrics": ["meter_kafka_partition_count"],
```
Comment (wankai123) [inline oap-server/server-starter/src/main/resources/ui-initialized-templates/kafka/kafka-cluster.json:21]: we should use MQE for queries in the new feature

**Review by wankai123 — CHANGES_REQUESTED** [review]

**Comment by wu-sheng** [comment]
Few tests still fail, I am not sure whether it is relative. Let's check again after you fix your mentor's comments.

**Comment by ZhuWang1112** [comment]
> Few tests still fail, I am not sure whether it is relative. Let's check again after you fix your mentor's comments.

I have reviewed my commits and did not find any changes to someone else's code. It might be a github network or system issue?

**Comment by wu-sheng** [comment]
GHA has unstable issues last week. I just approved your latest commit, let's see how it goes this time.

**Comment by wankai123** [comment]
It looks like your UI screenshot is not the latest, please update. And be consistent with the existing UI dashboard style.

File: .github/workflows/skywalking.yaml [hunk 8]
```diff
@@ -676,7 +676,9 @@ jobs:
           name: docker-images-11
           path: docker-images
       - name: Load docker images
-        run: find docker-images -name "*.tar" -exec docker load -i {} \;
+        run: |
+          find docker-images -name "*.tar" -exec docker load -i {} \;
+          find docker-images -name "*.tar" -exec rm {} \;
```
Comment (wu-sheng) [inline .github/workflows/skywalking.yaml:681]: @ZhuWang1112 We helped you to fix this. New downloads seem to be over the volume size limitation, which causes the failure.

Comment (ZhuWang1112) [reply .github/workflows/skywalking.yaml:681]: Really appreciate your help, and I will be mindful of the volume size limitation in the future.

**Review by wankai123 — APPROVED** [review]
LGTM

**Comment by wu-sheng** [comment]
@ZhuWang1112 Are you going to work on the showcase repo to make this available on the demo? If so, we already have elasticsearch deployment in the showcase I guess @wankai123 Is that possible to monitoring our own elasticsearch?

**Comment by ZhuWang1112** [comment]
> Are you going to work on the showcase repo to make this available on the demo?

Thanks for merging my PR! Yep, I'm on it - will add it to the showcase soon. Appreciate your support!

**Pull request merged**